qiskit_optimization/applications/SK_model.py

```diff
@@ -36,7 +36,7 @@ class SKModel(OptimizationApplication):
         https://arxiv.org/abs/1211.1094
     """
 
-    def __init__(self, n: int, rng: np.random.RandomState = np.random):
+    def __init__(self, n: int, rng: np.random.RandomState = np.random.RandomState(0)):
```

Comment:
Since this application is all around random behavior I guess its ideal that the random behavior here is independent. What we have otherwise is a random generator in algorithm_globals (in qiskit-terra utils folder now) that we use in general. This provides random behavior but we can see its behavior to have reproducible results. If this is independently random - but I see it truly isn't by default as you seed it. Might it be better to go over to the common generator we have - or if you want this independent allow an optional Generator or a seed to passed in. By default you can used np.random.default_rng() and if a seed is passed pass that to the default_rng() otherwise the user supply a Generator of their choosing. I think RandomState is more associated to the legacy random aspects of numpy.

Reply:
Thank you. Ok, I don't understand rng's in python.
I rea that there might be issues with reproducibility and RandomState is a good way how to deal with it https://stackoverflow.com/a/37224605
What do you suggest?

Reply:
I suggest using Random Generator instead of legacy Random State. The `RandomState` documentation says that
> This class should only be used if it is essential to have randoms that are identical to what would have been produced by previous versions of NumPy.

https://numpy.org/doc/stable/reference/random/legacy.html

Reply:
I agree that Generator based is the right way.

Either way what is the behavior you expect with a default object in the constructor like that. Do you expect it you instantiate two of these just number of sites, of the same value - since they appear to default with the same RandomState(0) that they will produce the same outcome. This is not the way initialization works unfortunately - arguably it might be better to default this to None and then if None internally create the object like that. This way the rng will be per instance as it stands now its shared across instances - which may not be self evident!

Reply:
Thank you @woodsp-ibm . I made the pseudo rng optional, and changed it to the current one. I hope there are no repeatability issues with it. Can you please check?